### DISCUSSION TITLE
Community Driver Usage and Testing

### DISCUSSION THREAD

**Discussion by jhaack** [VOLTTRON/volttron#2767] [Q&A]
We'd like to survey the community to better understand the current and potential usage of the different drivers and the types of devices they are being deployed against. This information will help focus community development efforts and provide relevant information on the status of each driver. If you would rather not relate information publicly, please email the PNNL development team directly.

Thank you

Driver List:

- Modbus 
- Modbus_TK
- BACNet 
- DNP3
- Ecobee
- Obix
- Radiothermostat
- Rainforesteagle
- Thermostat
- IEEE2030_5Agent 
- Other
- Custom Driver not in github

**Comment by RoshanLKini** [comment]
- Modbus
- Ecobee
- Custome Driver: ShiftedEnergyAPI, EcobeePubsusb
- BACnet

**Comment by craigpnnl** [comment]
I have used the 

- 2030_5
- dnp3

For dnp we tested points against Survalent using an application based off the dnp3 agent.  https://github.com/GRIDAPPSD/gridappsd-dnp3.  Eventually we want to update the agent in volttron based upon lessons learned developing that agent.

There is some missing things in the 2030_5 agent that I am going to need to work around.

**Comment by Bowriverstudio** [comment]
BACnet
Modbus

**Comment by ghrist** [comment]
I'm just starting research into all of them, and am mostly interested in:
CloudBus
BACnet
Modbus
MQTT
OPC-UA


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by craigpnnl** [reply]
I have not heard of cloudbus is this what you are talking about? https://cloud.spring.io/spring-cloud-bus/reference/html/

**Comment by acedrew** [comment]
- BACnet
 - ModBusTK
 - EthernetIP (to-be-oss)
 - OPC-UA (to-be-oss)
 - MQTT (to-be-oss, both directions)
 - Venstar Thermostat (to-be-oss)
 - CTA-2045 via MQTT (to-be-oss)
 - Obix (modified, to-be-oss)
 - OpenADR

**Comment by rlutes** [comment]
- BACnet 
- Modbus
- ModbusTk
- Radiothermostat
- Rainforesteagle
- Custom Driver to interact with TWT RTU

**Comment by ghrist** [comment]
No, that's Spring Cloud Bus. The CloudBUS I'm referring to is kind of new
and has no public resource yet.
I'm waiting on some documentation and will be sure to send a
relevant source on more information once it becomes available.

**Comment by amorphic** [comment]
- Modbus / ModbusTK
- LoRaWAN (possibly via a generic MQTT agent/driver)

**Comment by nanmin15** [comment]
- Open Charge Point Protocol (OCPP)

**Comment by Matammanjunath** [comment]
I know this is not the correct place to ask my question, but I failed to find the right place or location to ask this question. I heard about VOLTTRON a couple of days ago and trying to install and test this on a commercial building using Raspberry PIs. I tried installing the VOLTTRON on my Raspberry Pi by following the instructions [https://volttron.readthedocs.io/en/latest/introduction/platform-install.html](url) and [https://github.com/VOLTTRON/volttron](url). While installing this halfway, I am running into the issue. Presently, I am executing this line `vcfg rabbitmq single` and running into issue of `bash: vcfg: command not found`. I appreciate your help. Thanks 


![Screenshot 2023-02-28 181549](https://user-images.githubusercontent.com/41698531/222004724-c50c759e-02eb-4ccf-8fde-1e5fef25fffe.png)
